### PR TITLE
HWKALERTS-257 increase use of 400 return codes for illegal/bad args

### PR DESCRIPTION
**Pull request by jshaughn**
Simplify catch blocks by introducing a new utility to handle responses
on exceptions. As part of the changes do some code cleanup in the handlers:
- remove unnecessary log.isDebugEnabled clauses
- prefer log.debugf over log.debug
- remove unnecessary else-clauses after returns
- reorder response code annotations to be ascending numerically

### CODE REVIEW

File: hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/ActionPluginHandler.java [hunk 1]
```diff
@@ -70,18 +70,17 @@ public ActionPluginHandler() {
             response = String.class, responseContainer = "List")
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = "Successfully fetched list of actions plugins."),
+            @ApiResponse(code = 400, message = "Bad Request/Invalid Parameters.", response = ApiError.class),
```
Comment (lucasponce) [inline hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/ActionPluginHandler.java]: I think in this case perhaps a 400 is not necessary as this endpoint won't throw this one.
But no problem to maintain it as a pattern to not forget in other endpoints.

File: hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/AlertsHandler.java [hunk 2]
```diff
@@ -428,23 +411,17 @@ public Response addTags(
             @QueryParam("tags")
             final String tags) {
         try {
-            if (!isEmpty(alertIds) || isEmpty(tags)) {
+            if (!isEmpty(alertIds) && !isEmpty(tags)) {
```
Comment (lucasponce) [inline hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/AlertsHandler.java]: +1

File: hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/AlertsHandler.java [hunk 3]
```diff
@@ -467,23 +444,17 @@ public Response deleteTags(
             @QueryParam("tagNames")
             final String tagNames) {
         try {
-            if (!isEmpty(alertIds) || isEmpty(tagNames)) {
+            if (!isEmpty(alertIds) && !isEmpty(tagNames)) {
```
Comment (lucasponce) [inline hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/AlertsHandler.java]: +1

File: hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/EventsHandler.java [hunk 4]
```diff
@@ -193,23 +183,17 @@ public Response addTags(
             @QueryParam("tags")
             final String tags) {
         try {
-            if (!isEmpty(eventIds) || isEmpty(tags)) {
+            if (!isEmpty(eventIds) && !isEmpty(tags)) {
```
Comment (lucasponce) [inline hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/EventsHandler.java]: +1

File: hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/EventsHandler.java [hunk 5]
```diff
@@ -232,23 +216,17 @@ public Response deleteTags(
             @QueryParam("tagNames")
             final String tagNames) {
         try {
-            if (!isEmpty(eventIds) || isEmpty(tagNames)) {
+            if (!isEmpty(eventIds) && !isEmpty(tagNames)) {
```
Comment (lucasponce) [inline hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/EventsHandler.java]: +1

File: hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/ResponseUtil.java [hunk 6]
```diff
@@ -84,6 +86,25 @@ public static Response badRequest(String message) {
                 .entity(new ApiError(message)).type(APPLICATION_JSON_TYPE).build();
     }
 
+    public static Response onException(Exception e, Logger log) {
+        if (e instanceof NotFoundException || (null != e.getCause() && e.getCause() instanceof NotFoundException)) {
```
Comment (lucasponce) [inline hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/ResponseUtil.java]: Perhaps the expression can be simplified as 2nd part is also covered in next validation.

**Comment by lucasponce** [comment]
In general LGTM, merging to move on.

File: hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/ActionPluginHandler.java (hunk 1 above)
Comment (jshaughn) [inline hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/ActionPluginHandler.java]: I opted for consistency and added the 400 as a possible response given the new handler, even if at the current time we know the internals won't return IllegalArgumentException.  But by just listing 400 as a possibility it means the impl below the handler is free to return IllegalArgumentException without us having to change anything in the handler.  If you're totally opposed I can remove the 400 annotations, but I personally don't think it's an issue.

File: hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/ResponseUtil.java (hunk 6 above)
Comment (jshaughn) [inline hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/ResponseUtil.java]: +1

File: hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/ActionPluginHandler.java (hunk 1 above)
Comment (lucasponce) [inline hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/ActionPluginHandler.java]: It is fine, I have just left a note here just for the record.

**Pull request merged**